test/language/statements/class/subclass/class-definition-null-proto-missing-return-override.js

```diff
@@ -38,4 +38,4 @@ class Foo extends null {
 
 var foo = new Foo();
```

Comment:
Most implementations seem to throw on this right now, because 'this' is a use-before-decl in 'constructor () {}'. Perhaps because they treat the kind of the constructor as 'derived' and not 'base', as per #sec-ecmascript-function-objects-construct-argumentslist-newtarget step 5.

Can you confirm that this is an implementation gap and this isn't meant to throw here? Otherwise I agree the assert should be on Foo.prototype.